keras/applications/inception_v4.py

```diff
@@ -0,0 +1,375 @@
+    conv_bn1 = conv2d_bn(conv_bn1, 96, 3, 3, 1, 1, 'same')
+
+    conv_bn2 = conv2d_bn(x, 64, 1, 1, 1, 1, 'same')
+    conv_bn2 = conv2d_bn(conv_bn2, 96, 3, 3, 1, 1, 'same')
+    conv_bn2 = conv2d_bn(conv_bn2, 96, 3, 3, 1, 1, 'same')
+
+    p0 = AveragePooling2D((3, 3), strides=(1, 1), border_mode='same')(x)
+    conv_bn3 = conv2d_bn(p0, 96, 1, 1, 1, 1, 'same')
+
+    cat = merge([conv_bn0, conv_bn1, conv_bn2, conv_bn3],
+                mode='concat', concat_axis=channel_axis)
+    # cat1 = Activation('relu')(cat1)
+
+    return cat
+
+
+def inception_B(x, channel_axis):
+    conv_bn0 = conv2d_bn(x, 384, 1, 1, 1, 1, 'same')
+
+    conv_bn1 = conv2d_bn(x, 192, 1, 1, 1, 1, 'same')
+    conv_bn1 = conv2d_bn(conv_bn1, 224, 1, 7, 1, 1, 'same')
+    conv_bn1 = conv2d_bn(conv_bn1, 256, 7, 1, 1, 1, 'same')
+
+    conv_bn2 = conv2d_bn(x, 192, 1, 1, 1, 1, 'same')
+    conv_bn2 = conv2d_bn(conv_bn2, 192, 7, 1, 1, 1, 'same')
+    conv_bn2 = conv2d_bn(conv_bn2, 224, 1, 7, 1, 1, 'same')
+    conv_bn2 = conv2d_bn(conv_bn2, 224, 7, 1, 1, 1, 'same')
+    conv_bn2 = conv2d_bn(conv_bn2, 256, 1, 7, 1, 1, 'same')
+
+    p0 = AveragePooling2D((3, 3), strides=(1, 1), border_mode='same')(x)
+    conv_bn3 = conv2d_bn(p0, 128, 1, 1, 1, 1, 'same')
+
+    cat = merge([conv_bn0, conv_bn1, conv_bn2, conv_bn3],
+                mode='concat', concat_axis=channel_axis)
+
+    # cat1 = Activation('relu')(cat1)
+
+    return cat
+
+
+def inception_C(x, channel_axis):
+    conv_bn0 = conv2d_bn(x, 256, 1, 1, 1, 1, 'same')
+
+    conv_bn1 = conv2d_bn(x, 384, 1, 1, 1, 1, 'same')
+    conv_bn1_0 = conv2d_bn(conv_bn1, 256, 1, 3, 1, 1, 'same')
+    conv_bn1_1 = conv2d_bn(conv_bn1, 256, 3, 1, 1, 1, 'same')
+    conv_bn1 = merge([conv_bn1_0, conv_bn1_1],
+                     mode='concat', concat_axis=channel_axis)
+
+    conv_bn2 = conv2d_bn(x, 384, 1, 1, 1, 1, 'same')
+    conv_bn2 = conv2d_bn(conv_bn2, 448, 3, 1, 1, 1, 'same')
+    conv_bn2 = conv2d_bn(conv_bn2, 512, 1, 3, 1, 1, 'same')
+
+    conv_bn2_0 = conv2d_bn(conv_bn2, 256, 1, 3, 1, 1, 'same')
+    conv_bn2_1 = conv2d_bn(conv_bn2, 256, 3, 1, 1, 1, 'same')
+    conv_bn2 = merge([conv_bn2_0, conv_bn2_1],
+                     mode='concat', concat_axis=channel_axis)
+
+    p0 = AveragePooling2D((3, 3), strides=(1, 1), border_mode='same')(x)
+    conv_bn3 = conv2d_bn(p0, 256, 1, 1, 1, 1, 'same')
+
+    cat = merge([conv_bn0, conv_bn1, conv_bn2, conv_bn3],
+                mode='concat', concat_axis=channel_axis)
+
+    # cat1 = Activation('relu')(cat1)
+
+    return cat
+
+
+def auxiliary_logits(x):
+    """auxiliary head logits"""
+    x = AveragePooling2D((5, 5), strides=(3, 3))(x)
+    x = Convolution2D(128, 1, 1, 1, 1, 'same')(x)
+    x = Flatten()(x)
+    x = Dense(768, activation='relu')(x)
+    x = Dropout(0.2)(x)
+    x = Dense(1000, activation='softmax')(x)
+
+    return x
+
+
+def top_layer(x):
+    """final pooling and prediction"""
+    x = AveragePooling2D((8, 8), strides=(1, 1), border_mode='valid')(x)
```

Comment:
@kentsommer In keras document: it is said that 
strides: tuple of 2 integers, or None. Strides values. If None, it will default to pool_size.
So I think it is (8,8) if you remove it.